erpnext/accounts/doctype/sales_invoice/sales_invoice.py

```diff
@@ -1761,7 +1761,8 @@ def get_delivery_note_details(internal_reference):
 		filters={'parent': internal_reference})
 
 	for d in si_item_details:
-		so_item_map.setdefault(d.name, d.so_detail)
+		if d.so_detail:
+			so_item_map.setdefault(d.name, d.so_detail)
```

Comment:
`d.name` is supposed to be unique so `setdefault` is not required. If you just want to exclude `d.so_details = None` then it can be done in single dict comprehension too.



```python
so_item_map = {d.name: d.so_detail for d in si_item_details if d.so_detail}
```